src/hooks/useSync.ts

```diff
@@ -1,196 +1,59 @@
-}
-
-const findPantryDir = () => usePantry().prefix.parent()
-const findPantriesDir = () => findPantryDir().parent().join("pantries")
-
-let avoid_softlock = false
+export default async function() {
+  const logger = useLogger()
+  const pantry_dir = usePantry().prefix.parent()
 
-async function lock<T>(body: () => Promise<T>) {
-  if (avoid_softlock) throw new Error("aborting to prevent softlock")
-  avoid_softlock = true
+  logger.replace("syncing pantries…")
 
-  const { rid } = await Deno.open(findPantryDir().mkpath().string)
+  const { rid } = await Deno.open(pantry_dir.mkpath().string)
   await Deno.flock(rid, true)
 
   try {
-    return await body()
-  } finally {
-    //TODO if this gets stuck then nothing will work so need a handler for that
-    await Deno.funlock(rid)
-    Deno.close(rid)  // docs aren't clear if we need to do this or not
-    avoid_softlock = false
-  }
-}
-
-//TODO we have a better system in mind than git
-export async function install(logger: Logger): Promise<true | 'not-git' | 'noop' | 'deprecated'> {
-  const pantryDir = findPantryDir()
-  const pantriesDir = findPantriesDir()
-
-  if (usePantry().prefix.exists()) {
-    if (pantriesDir.exists()) return 'noop'
-    if (pantryDir.join('.git').exists()) return 'deprecated'
-
-    // FIXME in this case we have a downloaded pantry so we should install git
-    return 'not-git'
-  }
-
-  logger.replace("fetching pantries…")
+    //TODO if there was already a lock, just wait on it, don’t do the following stuff
 
-  try {
-    const git = await find_git()
-    const rv = await lock(async () => {
-      if (usePantry().prefix.exists()) return 'noop'
-      // ^^ another instance of tea did the install while we waited
+    const git_dir = pantry_dir.parent().join("pantries/teaxyz/pantry")
 
-      if (git) {
-        await clone(git)
-        return true
-      } else {
-        await unzip()
-        return 'not-git'
-      }
-    })
-    logger.replace("pantries init’d ⎷")
-    return rv
-  } catch (e) {
-    pantriesDir.rm({ recursive: true }) // leave us in a blank state
-    pantryDir.rm({ recursive: true })   // ^^
-    throw e
-  }
+    if (git_dir.join("HEAD").isFile()) {
+      await git("-C", git_dir, "fetch", "origin", "--force", "main:main")
+    } else {
+      await git("clone", "--bare", "--depth=1", "https://github.com/teaxyz/pantry", git_dir)
+    }
 
-  async function clone(git: Path) {
-    await run({
-      cmd: [
-        git, "clone",
-          "--bare", "--depth=1",
-          `https://github.com/teaxyz/pantry`,
-          pantriesDir.join("teaxyz").mkpath().join("pantry")
-      ]
-    })
-    await co(git)
-  }
+    await git("--git-dir", git_dir, "--work-tree", pantry_dir, "checkout", "--force")
 
-  async function unzip() {
-    //FIXME if we do this, we need to be able to convert it to a git installation later
-    //TODO use our tar if necessary
-    //TODO if we keep this then don’t store the files, just pipe to tar
+  } catch {
```

Comment:
if we log then every time the user does `--sync` it will log this error if the user has no `git` installed.

I agree that it's going to be shit to debug, but I'm hoping at this point we understand what is happening here well enough that we're safe.

the ideal flow is: if no git, download the tarball, then install git, then convert to a git local pantry. But that was more tricky to get right and I don't want bugs for gui launch